### PR TITLE
Subscribe button to survery

### LINKED ISSUE #37: Store a user's survey selection in the store

### LINKED ISSUE #38: User Story: First subscription routes to SurveyPage
As a user, I should be able to click subscribe on a crate and be brought to the survey page if it's my first time subscribing to a crate

### LINKED ISSUE #40: User Story: In SurveyPage, Clicking "Next Page" will load the next SurveyPage
As a user in a survey page, clicking the "Next Page" button will bring me to the next survey page with new images to select.

### LINKED ISSUE #43: User Story: SurveyPage displays images which become highlighted on-click
As a user, when brought to the survey page, I should see a selection of images that when clicked become highlighted to indicate that I have selected them.

### LINKED ISSUE #55: User Story: SurveyPage displays "Previous Page" button that is enabled if there is a previous page
As a user, I should see a previous page button that is enabled if there is a survey page to go back to (ie, if I'm on page 2 it is enabled, and if I'm on page 1 it is disabled

### PR DESCRIPTION
**Pull request by GreyMatteOr**
### This PRs additions:

- This PR creates a survey page template
  - posts to the db to set the user style to 'incomplete'
  - successfully requests items to display. Currently hard-coded to 'accessories', but changing to have dynamic requests should be trivial.
  - routes dynamically. path template: `/survey/:page` where `:page` is a number
  - changes item border-color to indicate selection status.

- Navigates to the survey page when the user style is `null`

 
### Reviewer starting point: 
- spin it up and login a user ('user@crate.com', 123456) and go to the url `http://localhost:3000/survey/1`
 
### How to manually test:  

- N/A, need to write tests.
 
### Relevant tickets: 

closes #43 
closes #40 
addresses #39
closes #38 
closes #37 
closes #55
 
### Screenshots:

Screen Shot 2020-12-10 at 4 33 00 PM

### TODO

- We need to make sure that all our images are either square or that our display handles different image sizes elegantly, otherwise we get distortion (as seen)
- Need to test to validate
- Need to decide order of survey and correctly fetch new data.

### CODE REVIEW

File: code/web/src/setup/routes/survey.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import List from '../../modules/survey/Survey/List'
 // Crate routes
 export default {
   survey: {
-    path: '/survey',
+    path: (page = ':page') => (`/survey/${ page }`),
```
Comment (MichaelEWalker87) [inline code/web/src/setup/routes/survey.js:7]: interesting, 
what is the ':page'

File: code/web/src/modules/survey/Survey/List.js [hunk 2]
```diff
@@ -59,15 +61,31 @@ class List extends PureComponent {
...
+        <Grid justifyCenter={true}>
+          <GridCell justifyCenter={true} style={{ padding: '2em', width: "50%"}}>
+            <Link to={surveyRoutes.survey.path( parseInt(this.props.match.params.page) - 1 )}>
+              <Button type="button" theme="secondary" disabled={this.props.match.params.page === '1'}>Back</Button>
+            </Link>
+          </GridCell>
+
+          <GridCell justifyRight={true} style={{ padding: '2em'}}>
+            <Link to={surveyRoutes.survey.path( parseInt(this.props.match.params.page) + 1 )}>
+              <Button type="button" theme="secondary" disabled={false}>Next</Button>
```
Comment (MichaelEWalker87) [inline code/web/src/modules/survey/Survey/List.js:85]: What does the theme do?

File: code/web/src/modules/survey/Survey/List.js [hunk 3]
```diff
@@ -59,15 +61,31 @@ class List extends PureComponent {
...
                   ))
                   : <EmptyMessage message="Something went wrong. Please go back and try again!" />
             }
           </GridCell>
         </Grid>
+
+        <Grid justifyCenter={true}>
+          <GridCell justifyCenter={true} style={{ padding: '2em', width: "50%"}}>
+            <Link to={surveyRoutes.survey.path( parseInt(this.props.match.params.page) - 1 )}>
+              <Button type="button" theme="secondary" disabled={this.props.match.params.page === '1'}>Back</Button>
```
Comment (MichaelEWalker87) [inline code/web/src/modules/survey/Survey/List.js:79]: This looks good, I will add the type to the button on my branch

File: code/web/src/modules/survey/Survey/List.js [hunk 4]
```diff
@@ -59,15 +61,31 @@ class List extends PureComponent {
               this.props.surveyItems.isLoading
                 ? <Loading/>
                 : this.props.surveyItems.surveyItems.length > 0
-                  ? this.props.surveyItems.surveyItems.map(item => (
-                    <div key={item.id} style={{ margin: '2em', float: 'left' }}>
+                  ? this.props.surveyItems.surveyItems.map((item, i) => (
+                    <div
+                      key={'item#'+i+'survey#'+this.props.match.params.page}
```
Comment (MichaelEWalker87) [inline code/web/src/modules/survey/Survey/List.js:66]: I am not 100% sure what this does 
key={'item#'+i+'survey#'+this.props.match.params.page}

File: code/web/src/modules/survey/Survey/Item.js [hunk 5]
```diff
@@ -27,22 +28,25 @@ class Item extends PureComponent {
...
-    </img>
+    return (
+      <img
+        src={routeImage + this.props.item.image}
+        style={{
+          height: '5em',
+          width: '5em',
+          borderStyle: 'solid',
+          borderWidth: '0.5em',
+          borderColor: this.state.selected ? primaryAccent : primary
```
Comment (MichaelEWalker87) [inline code/web/src/modules/survey/Survey/Item.js:45]: When looking at the mockup do you think that the colors would be easily visible?

**Review by MichaelEWalker87 — COMMENTED** [review]
Thank you for all your hard work. I had several questions below about things that got implemented for my personal understanding.

File: code/web/src/modules/survey/Survey/Item.js (hunk 5 above)
Comment (GreyMatteOr) [reply code/web/src/modules/survey/Survey/Item.js:45]: Yep! Good thought

File: code/web/src/modules/survey/Survey/List.js (hunk 4 above)
Comment (GreyMatteOr) [reply code/web/src/modules/survey/Survey/List.js:66]: React wants unique keys for Components in an Array. This now has a `key` of the shape

```Javascript
"item#1survey#1"
```

File: code/web/src/modules/survey/Survey/List.js (hunk 2 above)
Comment (GreyMatteOr) [reply code/web/src/modules/survey/Survey/List.js:85]: In `ui/Button.js` there's a `theme` prop referenced constantly. 'Secondary' is one of the pre-built options

File: code/web/src/setup/routes/survey.js (hunk 1 above)
Comment (GreyMatteOr) [reply code/web/src/setup/routes/survey.js:7]: This is used for dynamic routing.

`page = ':page' is a default for a parameter.

':page' is a string which will be the default route ====> `/survey/:page`

**Pull request merged**